features/mbedtls/platform/src/platform_alt.c

```diff
@@ -0,0 +1,59 @@
+ /*
+  *  platform_alt.c
+  *
+  *  Copyright (C) 2018, Arm Limited, All Rights Reserved
+  *  SPDX-License-Identifier: Apache-2.0
+  *
+  *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+  *  not use this file except in compliance with the License.
+  *  You may obtain a copy of the License at
+  *
+  *  http://www.apache.org/licenses/LICENSE-2.0
+  *
+  *  Unless required by applicable law or agreed to in writing, software
+  *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *  See the License for the specific language governing permissions and
+  *  limitations under the License.
+  *
+  */
+
+#include "mbedtls/platform.h"
+#if defined(MBEDTLS_PLATFORM_SETUP_TEARDOWN_ALT)
+
+static int reference_count = 0;
+
+int mbedtls_platform_setup( mbedtls_platform_context *ctx )
+{
+    int ret = 0;
+    if( ctx == NULL )
+        return ( -1 );
+
+    reference_count++;
+
+    if( reference_count == 1 )
+    {
+        /* call platform specific code to setup crypto driver*/
+        ret = crypto_platform_setup( &ctx->platform_impl_ctx );
```

Comment:
This makes `crypto_platform_setup()` part of the API for crypto drivers. That will need documenting in the Mbed OS Porting Guide, and also as an interface change will need wider review.

Is there anyway of avoiding this interface change? I'm not sure there is, but we should consider alternatives if we can.

Reply:
I agree it needs documenting in the handbook.

As for interface change, it's basically only a change in the existing name of the function ( `cc_platform_setup` renamed to `crypto+platform_setup`, and a bit of who is responsible of implementing CC initialization has changed). Although it is an API change, I believe it is a minor one
Unfortunately, I don't see any other way to have initialization of the platform,  in an opaque way, and make it portable to other APIs. ( we can have the whole `mbedtls_platform_setup()` functino implemented in driver code, but then we guideusers to implement the reference count , which is very common to all the drivers )